### PR TITLE
Dos plotting

### PR DESCRIPTION
**Pull request by kavanase**
Adding all the final bits of functionality discussed in #10; namely:
- DOS plotting alongside the unfolded band structures. This also sets the colours of ($lm$-decomposed) orbitals in the `sumo` DOS plot to match that of the projected band structure (if using the `plot-projections` option).
- Add `--atoms` option to just specify which atoms to project onto the band structure, without having to specify atomic indices.
- Some refactoring and consolidation of plotting options (i.e. changing `--vscale` to `--colour-norm` to make more intuitive, and using this instead of `--intensity` for `plot-projections` (same effect)).
- Show CLI help messages with `-h` option
- Major docs updates showing these features.
- Tests for all these features
- Update docstrings and some general code cleanup

Attaching a PDF of the updated NaBiS2 example which illustrates most of these features (some of the sphinx note/tip boxes don't render on my local, but do on the website version):
[Disordered NaBiS2 with atomic_orbital projections & DOS plotting — Easyunfold.pdf](https://github.com/SMTG-UCL/easyunfold/files/11981302/Disordered.NaBiS2.with.atomic_orbital.projections.DOS.plotting.Easyunfold.pdf)

### CODE REVIEW

**Comment by kavanase** [comment]
Also this means the `easyunfold 1.0.0` GH project is now completed. Ready to have the first major release @zhubonan?

**Comment by zhubonan** [comment]
Thanks for this! Looks fantastic. I will take a look and have it merged soon.

**Review by zhubonan — CHANGES_REQUESTED** [review]
Thanks a lot for this. Just some comments on the code structure and option naming. 

Also, I realised I have raised this PR #26 but not merged it lately. There I added an `--intensity` option which effectively changes the `vscale` by a factor of *1/intensity*. So `--intensity==2` would reduce the vscale by half and make the plot more *intense*. Can probably combine it into this PR as well?

**Comment by kavanase** [comment]
Ok will get to implementing these once I have time!

Just one thing about the `--intensity` point. This is what I suggested before in #10;
> (Seán): To me, the current setup of vscale is counter-intuitive (see https://smtg-ucl.github.io/easyunfold/cli/). It's a "scaling factor for the colour mapping", but decreasing it increases the colourmap intensity, whereas intuitively as a user I would've expected the opposite behaviour. Obvs a very minor thing, but I think it would be better to reverse this behaviour?

And your thoughts at the time were:
> (Bonan): Well I think the current behaviour is fine as it is the colour range scale, so using a small scale should increase the intensity. We can rename it as colour_scale to make it more informative.

If you're now happy to refactor this (now `colour-norm` to `intensity` – I agree!) then I'll go ahead with this! 😄

**Comment by zhubonan** [comment]
Thanks, I see, actually what I have in mind is to have both `--vscale` and `--intensity` options available for now. 

The `--vscale` behaves as before (*value scale*), so setting it to be less than 1 will make the colour more intensity. And the new option `--intensity` also affects the colour but in the opposite way, larger value increase the intensity of the colours. This is probably more intuitive. Compared to renaming `--vscale` to `--colour-norm` or `--intensity` with new behaviour, this approach ensures the backward compatibility. 

We can issue deprecation warnings for `--vscale` if need in the future. What do you think?

**Comment by kavanase** [comment]
Yeah I just think the average user is not so familiar with `vscale`. We can improve this with a better help message explaining it, but as you said (and I was saying before), `intensity` is definitely a lot more intuitive...
We could keep both options as you say, but at this point is backwards-compatibility really important? Like you, myself and Adair have been the main users so far, and maybe some others from the group, but hasn't been massively advertised / grown the user/dependency base yet, so I'd suggest that this isn't really important right now?

There's already a lot of options for the CLI plotting, so I would think best to not have any redundant ones like this, and also two options doing the exact opposite thing could also just confuse people? (The `intensity` and `vscale` options confused me at first when I first came back to look at the code a few weeks ago)

**Comment by zhubonan** [comment]
Right, I think maitaining backward compatible is important. It is probably one of the things prevant people from using a code under active development (and complain about). Since we have opted in sematic versioning so users would expect backward compatible behaviour. I agree that the help message should definitely be updated to more informative as you said (which this PR already improved 🚀). Hopefully we are  gaining more users soon (after MC16 😉 ). 

Regarding the number of options - yes I think they are a bit too many at the moment. It should be possible to use a file to provide them instead with, which works by  internally modified the default values. I will look into that after the PR is merged.

See also:https://click.palletsprojects.com/en/7.x/commands/#overriding-defaults

**Comment by kavanase** [comment]
Hi @zhubonan! Sorry I had a few deadlines this week so just getting to this now. 😬 

Ok fair enough, though we could also just bump to a major version update seeing as there's some major upgrades that have been added, but your call! 😃 

For the CLI options, I think it's nice to have a good bit of customisation control as we have now, but also trying to maximise the control while minimising the number of options (and having a good choice of defaults so the user shouldn't need to use a lot of the options in most cases), which was why I was against the redundant options. For something like this, I would have thought that any users relying on the old CLI (which I don't think there is any) would be fixed to that version, or just have installed that version on their local, to ensure compatibility. And if they have it installed locally and are just updating, it seems like something one would figure out quite quickly after getting e.g. `colour_norm` option not found, and realising `intensity` had been added instead. 

What you say about a `config.yml` option I think would be a nice addition (though as a separate CLI option would be best for user-friendliness I think, rather than replacing the CLI options). We have this in `ShakeNBreak`, e.g.:
image

https://github.com/SMTG-UCL/ShakeNBreak/blob/main/SnB_input_files/example_generate_config.yaml
- Worth adding as a `TODO` for after this PR & JOSS?

Will try get to your comments on the PR and wrap this up ASAP!

File: easyunfold/cli.py
Comment (kavanase) [reply easyunfold/cli.py:162]: So the reason I moved it from here was that it didn't show up in the `easyunfold unfold plot -h` message (where most people would be looking for something like this), and also had to be given in an odd format as 
```
easyunfold unfold --mpl-style-file X.mplstyle plot --ncl ...
```
(i.e. before the `plot` etc subcommands). 
I've updated this to make it a decorator now to avoid the code duplication but still show it as an option in the help messages for `effective_mass`, `plot` and `plot-projections` ✅

File: easyunfold/unfold.py
Comment (kavanase) [reply easyunfold/unfold.py:1101]: Done ✅

File: docs/examples/example_mgo.md
Comment (kavanase) [reply docs/examples/example_mgo.md:31]: Yeah good spot, the `PROCAR` format differs but I thought this was fine with the current parser, but from rechecking it breaks with `LORBIT >= 12`. I've now updated this and tested it works fine for all `LORBIT` = `11`, `12`, `13` and `14`, and added a test for this

File: easyunfold/cli.py
Comment (kavanase) [reply easyunfold/cli.py:432]: Yep good point, should've done this from the start. Was just from the way I started coding it (for the projected plot first, then realising it could go in the other and thinking the handling would differ significantly etc etc). Done now! 😃

File: easyunfold/plotting.py
Comment (kavanase) [reply easyunfold/plotting.py:141]: Also good spot. Same excuses as the other for not doing this from the start 🤦 
Done now! 😃

File: easyunfold/cli.py
Comment (kavanase) [reply easyunfold/cli.py:220]: 👍

**Comment by kavanase** [comment]
@zhubonan all requested changes & updates have now been implemented 😃 

I've also merged in #26 as requested, so I'll close that PR now. Let's get this merged and spread the word! 🚀

**Review by zhubonan — APPROVED** [review]
Thanks for this. All look good to me. 🚀

**Pull request merged**
